stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -120,21 +116,37 @@ extension ${Self}: CustomDebugStringConvertible {
   /// This property has the same value as the `description` property, except
   /// that NaN values are printed in an extended format.
   public var debugDescription: String {
-    var (buffer, length) = _float${bits}ToString(self, debug: true)
-    return unsafe buffer.withBytes { (bufferPtr) in
-      unsafe String._fromASCII(
-        UnsafeBufferPointer(start: bufferPtr, count: length))
+    if #available(SwiftStdlib 6.2, *) {
```

Comment:
@tshortli  @al45tair  --  In my local testing, it appears that this check results in a call to `__isPlatformVersionAtLeast` which takes a full 1/3 of the run time for `debugDescription`.  This shouldn't be necessary at all -- the implementation here has no dependencies outside of the standard library that we're part of.  How can we eliminate this?

Reply:
Local stdlib builds set the deployment target to something way too old, see 144244211

Reply:
You could also use `StdlibDeploymentTarget 6.2` here (but as before, you might end up having to move more things to `StdlibDeploymentTarget 6.2` as a result). That will get rid of the check, because when you build locally, you'll be testing for the target you're building for.

That would let you test the performance sensibly here, I think.

Reply:
You can override the default deployment target when building locally by passing something like `--darwin-deployment-version-osx 26` to `--build-script`, if you just want to look at performance locally.

Reply:
> You could also use StdlibDeploymentTarget 6.2 here (but as before, you might end up having to move more things to StdlibDeploymentTarget 6.2 as a result). 

Nope.  That doesn't work for `InlineArray`, at least not with a stock `build-script -rA` invocation, since "value generics are only available on macOS 26."  I discussed this in #83710.  If I landed that, then I could duplicate `InlineArray` to make `internal _InlineArray` which would not need any availability restrictions at all.  Using the same pattern to create `internal _MutableSpan` would probably just eliminate all of the availability checks in this code.

I've not experimented with `_unreachable()` yet.

I'm playing with build-script incantations to see what other options exist.

Reply:
> ... something like --darwin-deployment-version-osx 26 to build-script, if you just want to look at performance locally

Sure, but that still misses the point: Nothing in this code requires macOS 26.  Everything it needs is in libswiftCore. The runtime host version is irrelevant.

Reply:
It would let you look at performance locally without the version checks getting in the way. It doesn’t solve the larger issue, of course.